cloud/services/publicips/publicips.go

```diff
@@ -31,11 +31,20 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	for _, ip := range s.Scope.PublicIPSpecs() {
 		s.Scope.V(2).Info("creating public IP", "public ip", ip.Name)
 
+		// only set DNS properties if there is a DNS name specified
 		addressVersion := network.IPv4
 		if ip.IsIPv6 {
 			addressVersion = network.IPv6
 		}
 
+		var dnsSettings *network.PublicIPAddressDNSSettings
+		if ip.DNSName != "" {
+			dnsSettings = &network.PublicIPAddressDNSSettings{
+				DomainNameLabel: to.StringPtr(strings.ToLower(ip.Name)),
+				Fqdn:            to.StringPtr(ip.DNSName),
```

Comment:
food for thought (for later) - after taking a closer look at this I'm not sure we need to specify the Fqdn since it would just default to DomainNameLabel + Azure regionalized DNS zone (unless we were trying to support custom DNS)

```
// PublicIPAddressDNSSettings contains FQDN of the DNS record associated with the public IP address.
type PublicIPAddressDNSSettings struct {
	// DomainNameLabel - Gets or sets the Domain name label.The concatenation of the domain name label and the regionalized DNS zone make up the fully qualified domain name associated with the public IP address. If a domain name label is specified, an A DNS record is created for the public IP in the Microsoft Azure DNS system.
	DomainNameLabel *string `json:"domainNameLabel,omitempty"`
	// Fqdn - Gets the FQDN, Fully qualified domain name of the A DNS record associated with the public IP. This is the concatenation of the domainNameLabel and the regionalized DNS zone.
	Fqdn *string `json:"fqdn,omitempty"`
	// ReverseFqdn - Gets or Sets the Reverse FQDN. A user-visible, fully qualified domain name that resolves to this public IP address. If the reverseFqdn is specified, then a PTR DNS record is created pointing from the IP address in the in-addr.arpa domain to the reverse FQDN.
	ReverseFqdn *string `json:"reverseFqdn,omitempty"`
}
```